source/DSCResources/DSC_CMAssetIntelligencePoint/DSC_CMAssetIntelligencePoint.psm1

```diff
@@ -0,0 +1,446 @@
+            }
+        }
+        elseif ($state.Ensure -eq 'Present')
+        {
+            Write-Verbose -Message ($script:localizedData.RemoveAPRole -f $SiteServerName)
+            Remove-CMAssetIntelligenceSynchronizationPoint -SiteSystemServerName $SiteServerName -SiteCode $SiteCode
+        }
+    }
+    catch
+    {
+        throw $_
+    }
+    finally
+    {
+        Set-Location -Path "$env:temp"
+    }
+}
+
+<#
+    .SYNOPSIS
+        This will test the desired state.
+
+    .PARAMETER SiteCode
+        Specifies a site code for the Configuration Manager site that manages the system role for the asset intelligence point.
+
+    .PARAMETER SiteServerName
+        Specifies the Site Server to install or configure the role on.
+
+    .PARAMETER CertificateFile
+        Specifies the path to a System Center Online authentication certificate (.pfx) file.
+        If used, this must be in UNC format. Local paths are not allowed.
+        Mutually exclusive with the CertificateFile Parameter.
+
+    .PARAMETER Schedule
+        Specifies when the asset intelligence catalog is synchronized.
+
+    .PARAMETER Enable
+        Specifies whether the installed asset intelligence role is enabled or disabled.
+
+    .PARAMETER EnableSynchronization
+        Specifies whether to synchronize the asset intelligence catalog.
+
+    .PARAMETER RemoveCertificate
+        Specifies whether to remove a configured certificate file.
+        Mutually exclusive with the CertificateFile Parameter.
+
+    .PARAMETER Ensure
+        Specifies whether the asset intelligence synchronization point is present or absent.
+
+    .Notes
+        This role must only be installed on top-level site of the hierarchy.
+#>
+
+function Test-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Boolean])]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [String]
+        $SiteCode,
+
+        [Parameter(Mandatory = $true)]
+        [String]
+        $SiteServerName,
+
+        [Parameter()]
+        [String]
+        $CertificateFile,
+
+        [Parameter()]
+        [Microsoft.Management.Infrastructure.CimInstance]
+        $Schedule,
+
+        [Parameter()]
+        [Boolean]
+        $Enable,
+
+        [Parameter()]
+        [Boolean]
+        $EnableSynchronization,
+
+        [Parameter()]
+        [Boolean]
+        $RemoveCertificate,
+
+        [Parameter()]
+        [ValidateSet('Present','Absent')]
+        [String]
+        $Ensure = 'Present'
+    )
+
+    Import-ConfigMgrPowerShellModule -SiteCode $SiteCode
+    Set-Location -Path "$($SiteCode):\"
+    $state = Get-TargetResource -SiteCode $SiteCode -SiteServerName $SiteServerName
+    $result = $true
+
+    if ($Ensure -eq 'Present')
+    {
+        if ($state.Ensure -eq 'Absent')
+        {
+            Write-Verbose -Message ($script:localizedData.APNotInstalled -f $SiteServerName)
+            $result = $false
+        }
+
```

Comment:
maybe add an else here if the role is not installed, no need to check the rest of the params

Reply:
since adding an else statement could also do a Write-Verbose message stating that SiteSystemServer parameter will be ignored as the role is already installed.

Reply:
added else statement and went ahead and added the write-verbose as well.